test/powershell/Modules/Microsoft.PowerShell.Core/RemoteImportModule.Tests.ps1

```diff
@@ -0,0 +1,111 @@
+Describe "Remote import-module tests" -Tags 'Feature','RequireAdminOnWindows' {
+
+    BeforeAll {
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        $modulePath = "$testdrive\Modules\TestImport"
+        if (!$IsWindows) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        } else {
+            $pssession = New-RemoteSession
+            Invoke-Command -Session $pssession -ScriptBlock { $env:PSModulePath += ";${using:testdrive}" }
+            # pending https://github.com/PowerShell/PowerShell/issues/4819
+            # $cimsession = New-RemoteSession -CimSession
+            $null = New-Item -ItemType Directory -Path $modulePath
+            Set-Content -Path $modulePath\testimport.psm1 -Value "function test-hello { 'world' }"
+            New-ModuleManifest -Path $modulePath\testimport.psd1 -ModuleVersion 1.2.3 -RootModule testimport.psm1 -FunctionsToExport "test-hello" `
+                -HelpInfoUri "https://help" -Guid (New-Guid)
+        }
+    }
+
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        if ($IsWindows) {
+            $pssession | Remove-PSSession -ErrorAction SilentlyContinue
+        }
+    }
+
+    BeforeEach {
+        Remove-Module TestImport -Force -ErrorAction SilentlyContinue
+    }
+
+    It "Import-Module can be called as an API with '<parameter>' = '<value>'" -TestCases @(
+        @{parameter = "Global"             ; value = $true},
+        @{parameter = "Global"             ; value = $false},
+        @{parameter = "Prefix"             ; value = "Hello"},
+        @{parameter = "Name"               ; value = "foo","bar"},
+        @{parameter = "FullyQualifiedName" ; value = @{ModuleName='foo';RequiredVersion='0.0'},@{ModuleName='bar';RequiredVersion='1.1'}},
+        @{parameter = "Assembly"           ; script = { [System.AppDomain]::CurrentDomain.GetAssemblies() | Select-Object -First 2 }}

```

Comment:
Can't the Assembly array be pre-computed and passed in as a "value" parameter?  Then you wouldn't need the special case for "script" parameter.

Reply:
My thinking is that if we add new test cases here, we can reuse `script` even though it's only used once here.  Also, I don't like having variables defined at the top of the script in a `BeforeAll` statement if it's only used once for a specific `It`, prefer to have it closer to the code where it's being used.